// devextreme-angular/esm2022/ui/diagram/nested/tab-dxi.mjs
/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
/* tslint:disable:max-line-length */
import { Component, NgModule, Host, SkipSelf, Input, ContentChildren, QueryList } from '@angular/core';
import { DxIntegrationModule, NestedOptionHost, } from 'devextreme-angular/core';
import { CollectionNestedOption } from 'devextreme-angular/core';
import { PROPERTY_TOKEN_tabs } from 'devextreme-angular/core/tokens';
import { PROPERTY_TOKEN_commands, PROPERTY_TOKEN_groups, } from 'devextreme-angular/core/tokens';
import * as i0 from "@angular/core";
import * as i1 from "devextreme-angular/core";
export class DxiDiagramTabComponent extends CollectionNestedOption {
    set _commandsContentChildren(value) {
        this.setChildren('commands', value);
    }
    set _groupsContentChildren(value) {
        this.setChildren('groups', value);
    }
    get commands() {
        return this._getOption('commands');
    }
    set commands(value) {
        this._setOption('commands', value);
    }
    get groups() {
        return this._getOption('groups');
    }
    set groups(value) {
        this._setOption('groups', value);
    }
    get title() {
        return this._getOption('title');
    }
    set title(value) {
        this._setOption('title', value);
    }
    get _optionPath() {
        return 'tabs';
    }
    constructor(parentOptionHost, optionHost) {
        super();
        parentOptionHost.setNestedOption(this);
        optionHost.setHost(this, this._fullOptionPath.bind(this));
    }
    ngOnDestroy() {
        this._deleteRemovedOptions(this._fullOptionPath());
    }
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxiDiagramTabComponent, deps: [{ token: i1.NestedOptionHost, host: true, skipSelf: true }, { token: i1.NestedOptionHost, host: true }], target: i0.ɵɵFactoryTarget.Component });
    /** @nocollapse */ static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: DxiDiagramTabComponent, isStandalone: true, selector: "dxi-diagram-tab", inputs: { commands: "commands", groups: "groups", title: "title" }, providers: [
            NestedOptionHost,
            {
                provide: PROPERTY_TOKEN_tabs,
                useExisting: DxiDiagramTabComponent,
            }
        ], queries: [{ propertyName: "_commandsContentChildren", predicate: PROPERTY_TOKEN_commands }, { propertyName: "_groupsContentChildren", predicate: PROPERTY_TOKEN_groups }], usesInheritance: true, ngImport: i0, template: '', isInline: true, styles: [""], dependencies: [{ kind: "ngmodule", type: DxIntegrationModule }] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxiDiagramTabComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dxi-diagram-tab', standalone: true, template: '', imports: [DxIntegrationModule], providers: [
                        NestedOptionHost,
                        {
                            provide: PROPERTY_TOKEN_tabs,
                            useExisting: DxiDiagramTabComponent,
                        }
                    ] }]
        }], ctorParameters: () => [{ type: i1.NestedOptionHost, decorators: [{
                    type: SkipSelf
                }, {
                    type: Host
                }] }, { type: i1.NestedOptionHost, decorators: [{
                    type: Host
                }] }], propDecorators: { _commandsContentChildren: [{
                type: ContentChildren,
                args: [PROPERTY_TOKEN_commands]
            }], _groupsContentChildren: [{
                type: ContentChildren,
                args: [PROPERTY_TOKEN_groups]
            }], commands: [{
                type: Input
            }], groups: [{
                type: Input
            }], title: [{
                type: Input
            }] } });
export class DxiDiagramTabModule {
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxiDiagramTabModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.12", ngImport: i0, type: DxiDiagramTabModule, imports: [DxiDiagramTabComponent], exports: [DxiDiagramTabComponent] });
    /** @nocollapse */ static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxiDiagramTabModule, imports: [DxiDiagramTabComponent] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DxiDiagramTabModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        DxiDiagramTabComponent
                    ],
                    exports: [
                        DxiDiagramTabComponent
                    ],
                }]
        }] });
