// devextreme-schematics/src/utility/typescript-resolver.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.resolveTypeScript = void 0;
const child_process_1 = require("child_process");
const path = __importStar(require("path"));
const fs = __importStar(require("fs"));
/**
 * Resolves TypeScript with a 3-level fallback strategy:
 * 1. Project search - look in the user's project node_modules
 * 2. Global search - look in global node_modules
 * 3. Temporary install - use npx to temporarily install typescript
 */
function resolveTypeScript() {
    const errors = [];
    try {
        const projectTs = tryResolveFromProject();
        if (projectTs) {
            return {
                ts: projectTs,
                resolutionMethod: 'project',
                errors: []
            };
        }
    }
    catch (err) {
        errors.push(`Project resolution failed: ${(err === null || err === void 0 ? void 0 : err.message) || err}`);
    }
    try {
        const globalTs = tryResolveFromGlobal();
        if (globalTs) {
            return {
                ts: globalTs,
                resolutionMethod: 'global',
                errors: []
            };
        }
    }
    catch (err) {
        errors.push(`Global resolution failed: ${(err === null || err === void 0 ? void 0 : err.message) || err}`);
    }
    try {
        const tempTs = tryResolveViaTemporaryInstall();
        if (tempTs) {
            return {
                ts: tempTs,
                resolutionMethod: 'temporary',
                errors: []
            };
        }
    }
    catch (err) {
        errors.push(`Temporary install failed: ${(err === null || err === void 0 ? void 0 : err.message) || err}`);
    }
    return {
        ts: null,
        resolutionMethod: null,
        errors
    };
}
exports.resolveTypeScript = resolveTypeScript;
function tryResolveFromProject() {
    var _a;
    const searchPaths = [
        process.cwd(),
        path.dirname(((_a = require.main) === null || _a === void 0 ? void 0 : _a.filename) || ''),
    ];
    for (const searchPath of searchPaths) {
        try {
            const tsPath = require.resolve('typescript', { paths: [searchPath] });
            // tslint:disable-next-line:no-var-requires
            return require(tsPath);
        }
        catch (_b) {
            // Continue to next path
        }
    }
    return null;
}
function tryResolveFromGlobal() {
    try {
        const globalPath = (0, child_process_1.execSync)('npm root -g', { encoding: 'utf8' }).trim();
        const typescriptPath = path.join(globalPath, 'typescript');
        if (fs.existsSync(typescriptPath)) {
            // tslint:disable-next-line:no-var-requires
            return require(typescriptPath);
        }
    }
    catch (_a) {
        // Fall through
    }
    return null;
}
function tryResolveViaTemporaryInstall() {
    try {
        const tmpDir = path.join(require('os').tmpdir(), 'devextreme-schematics-ts-' + Date.now());
        fs.mkdirSync(tmpDir, { recursive: true });
        const packageJson = {
            name: 'temp-typescript-resolver',
            version: '1.0.0',
            dependencies: {
                typescript: 'latest'
            }
        };
        fs.writeFileSync(path.join(tmpDir, 'package.json'), JSON.stringify(packageJson, null, 2));
        (0, child_process_1.execSync)('npm install --silent --no-progress', {
            cwd: tmpDir,
            stdio: 'ignore',
            timeout: 30000
        });
        const typescriptPath = path.join(tmpDir, 'node_modules', 'typescript');
        if (fs.existsSync(typescriptPath)) {
            // tslint:disable-next-line:no-var-requires
            const ts = require(typescriptPath);
            setTimeout(() => {
                try {
                    fs.rmSync(tmpDir, { recursive: true, force: true });
                }
                catch (_a) {
                    // Ignore cleanup errors
                }
            }, 5000);
            return ts;
        }
    }
    catch (_a) {
        // Fall through
    }
    return null;
}
