// devextreme-schematics/src/add-view/index_spec.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const testing_1 = require("@angular-devkit/schematics/testing");
const core_1 = require("@angular-devkit/core");
const path = __importStar(require("path"));
const collectionPath = path.join(__dirname, '../collection.json');
describe('view', () => {
    const appOptions = {
        name: 'testApp',
        projectRoot: '',
        inlineStyle: false,
        inlineTemplate: false,
        routing: true,
        style: 'css',
        skipTests: false,
        skipPackageJson: false,
        standalone: false
    };
    const workspaceOptions = {
        name: 'workspace',
        version: '6.0.0'
    };
    const componentOptions = {
        name: 'test',
        inlineStyle: false,
        inlineTemplate: false,
        changeDetection: 'Default',
        styleext: 'css',
        skipImport: true,
        module: undefined,
        export: false,
        project: 'testApp'
    };
    const angularSchematicsCollection = require.resolve('../../node_modules/@schematics/angular/collection.json');
    const schematicRunner = new testing_1.SchematicTestRunner('@schematics/angular', angularSchematicsCollection);
    let appTree;
    beforeEach(() => __awaiter(void 0, void 0, void 0, function* () {
        appTree = yield schematicRunner.runSchematic('workspace', workspaceOptions);
        appTree = yield schematicRunner.runSchematic('application', appOptions, appTree);
    }));
    it('should create new view', () => __awaiter(void 0, void 0, void 0, function* () {
        const runner = new testing_1.SchematicTestRunner('schematics', collectionPath);
        const componentName = componentOptions.name;
        let tree = yield runner.runSchematic('add-layout', { layout: 'side-nav-outer-toolbar' }, appTree);
        tree = yield runner.runSchematic('add-view', componentOptions, appTree);
        expect(tree.files).toContain('/src/app/pages/test/test.component.ts');
        expect(tree.files).toContain('/src/app/pages/test/test.component.html');
        const contentHTML = tree.readContent('/src/app/pages/test/test.component.html');
        const contentTS = tree.readContent('/src/app/pages/test/test.component.ts');
        expect(contentHTML).toMatch(/<h2>Test<\/h2>/);
        expect(contentTS).toContain(`selector: 'app-${componentName}'`);
        expect(contentTS).toContain(`templateUrl: './${componentName}.component.html'`);
        expect(contentTS).toContain(`styleUrl: './${componentName}.component.css'`);
        expect(contentTS).toContain('standalone: true');
        expect(contentTS).toContain(`export class ${core_1.strings.classify((0, core_1.basename)((0, core_1.normalize)(componentName)))}Component`);
    }));
    it('should add view to default routing module', () => __awaiter(void 0, void 0, void 0, function* () {
        const options = Object.assign(Object.assign({}, componentOptions), { addRoute: true });
        const runner = new testing_1.SchematicTestRunner('schematics', collectionPath);
        let tree = yield runner.runSchematic('add-layout', { layout: 'side-nav-outer-toolbar' }, appTree);
        tree = yield runner.runSchematic('add-view', options, tree);
        tree = yield runner.runSchematic('add-view', Object.assign(Object.assign({}, options), { name: 'test2' }), tree);
        const moduleContent = tree.readContent('/src/app/app.routes.ts');
        expect(moduleContent).toContain(`path: 'pages/test2',\n    component: Test2Component,\n    canActivate: [ AuthGuardService ]`);
        expect(moduleContent).toContain(`path: 'pages/test',\n    component: TestComponent,\n    canActivate: [ AuthGuardService ]`);
    }));
    it('should add view to other routing module', () => __awaiter(void 0, void 0, void 0, function* () {
        const options = Object.assign(Object.assign({}, componentOptions), { addRoute: true, module: 'test/test-routing' });
        const runner = new testing_1.SchematicTestRunner('schematics', collectionPath);
        let tree = yield runner.runExternalSchematic('@schematics/angular', 'module', {
            name: 'test',
            routing: true,
            project: 'testApp'
        }, appTree);
        tree = yield runner.runSchematic('add-layout', {
            layout: 'side-nav-outer-toolbar',
            project: 'testApp',
            name: 'test'
        }, tree);
        tree = yield runner.runSchematic('add-view', options, tree);
        const moduleContent = tree.readContent('/src/app/test/test-routing.module.ts');
        expect(moduleContent).toMatch(/component: TestComponent/);
        expect(moduleContent).toMatch(/path: 'pages\/test'/);
        expect(moduleContent).toContain('canActivate: [ AuthGuardService ]');
    }));
    it('should add view to navigation', () => __awaiter(void 0, void 0, void 0, function* () {
        const runner = new testing_1.SchematicTestRunner('schematics', collectionPath);
        let tree = yield runner.runSchematic('add-layout', { layout: 'side-nav-outer-toolbar' }, appTree);
        tree = yield runner.runSchematic('add-view', componentOptions, tree);
        componentOptions.name = 'some test';
        componentOptions.icon = 'home';
        tree = yield runner.runSchematic('add-view', componentOptions, tree);
        const moduleContent = tree.readContent('/src/app/app-navigation.ts');
        expect(moduleContent).toMatch(/text: 'Some Test'/);
        expect(moduleContent).toMatch(/icon: 'home'/);
        expect(moduleContent).toMatch(/text: 'Test'/);
        expect(moduleContent).toMatch(/icon: 'folder'/);
        expect(moduleContent).toContain(`navigation = [
  {
    text: 'Test',
    path: '/pages/test',
    icon: 'folder'
  },
  {
    text: 'Some Test',
    path: '/pages/some-test',
    icon: 'home'
  }
];`);
        const pageContent = tree.readContent('/src/app/pages/some-test/some-test.component.html');
        expect(pageContent).toMatch(/<h2>Some Test<\/h2>/);
    }));
    it('should create new view with path', () => __awaiter(void 0, void 0, void 0, function* () {
        const runner = new testing_1.SchematicTestRunner('schematics', collectionPath);
        componentOptions.name = 'folder/test';
        let tree = yield runner.runSchematic('add-layout', { layout: 'side-nav-outer-toolbar' }, appTree);
        tree = yield runner.runSchematic('add-view', componentOptions, appTree);
        expect(tree.files).toContain('/src/app/folder/test/test.component.ts');
        expect(tree.files).toContain('/src/app/folder/test/test.component.html');
    }));
});
