// devextreme/cjs/viz/core/renderers/renderer.js
/**
 * DevExtreme (cjs/viz/core/renderers/renderer.js)
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 * Read about DevExtreme licensing here: https://js.devexpress.com/Licensing/
 */
"use strict";
Object.defineProperty(exports, "ArcSvgElement", {
    enumerable: true,
    get: function() {
        return _renderer.ArcSvgElement
    }
});
Object.defineProperty(exports, "PathSvgElement", {
    enumerable: true,
    get: function() {
        return _renderer.PathSvgElement
    }
});
Object.defineProperty(exports, "RectSvgElement", {
    enumerable: true,
    get: function() {
        return _renderer.RectSvgElement
    }
});
Object.defineProperty(exports, "Renderer", {
    enumerable: true,
    get: function() {
        return _renderer.Renderer
    }
});
Object.defineProperty(exports, "SvgElement", {
    enumerable: true,
    get: function() {
        return _renderer.SvgElement
    }
});
Object.defineProperty(exports, "TextSvgElement", {
    enumerable: true,
    get: function() {
        return _renderer.TextSvgElement
    }
});
Object.defineProperty(exports, "getBackup", {
    enumerable: true,
    get: function() {
        return _renderer.getBackup
    }
});
Object.defineProperty(exports, "getFuncIri", {
    enumerable: true,
    get: function() {
        return _renderer.getFuncIri
    }
});
Object.defineProperty(exports, "processHatchingAttrs", {
    enumerable: true,
    get: function() {
        return _renderer.processHatchingAttrs
    }
});
Object.defineProperty(exports, "refreshPaths", {
    enumerable: true,
    get: function() {
        return _renderer.refreshPaths
    }
});
var _renderer = require("../../../__internal/viz/core/renderers/renderer");
