// devextreme/cjs/__internal/grids/new/grid_core/editing/utils.js
/**
 * DevExtreme (cjs/__internal/grids/new/grid_core/editing/utils.js)
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 * Read about DevExtreme licensing here: https://js.devexpress.com/Licensing/
 */
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.defaultSetFieldValue = exports.PendingPromises = void 0;
const defaultSetFieldValue = function(newData, value) {
    const {
        dataField: dataField
    } = this;
    if (!dataField) {
        return
    }
    newData[dataField] = value
};
exports.defaultSetFieldValue = defaultSetFieldValue;
class PendingPromises {
    constructor() {
        this.promises = new Set
    }
    waitForAll() {
        return Promise.all([...this.promises])
    }
    add(p) {
        this.promises.add(p);
        p.then((() => {
            this.promises.delete(p)
        }));
        return p
    }
}
exports.PendingPromises = PendingPromises;
