// devextreme-angular/fesm2022/devextreme-angular-common.mjs
export { Guid, config, setTemplateEngine } from 'devextreme/common';
import { AIIntegration } from 'devextreme/common/ai-integration';
import { registerGradient, registerPattern } from 'devextreme/common/charts';
import { TransitionExecutor, animationPresets, cancelAnimationFrame, fx, requestAnimationFrame } from 'devextreme/common/core/animation';
import { getTimeZones, hideTopOverlay, initMobileViewport } from 'devextreme/common/core/environment';
import { off, on, one, trigger } from 'devextreme/common/core/events';
import { formatDate, formatMessage, formatNumber, loadMessages, locale, parseDate, parseNumber } from 'devextreme/common/core/localization';
import { ArrayStore, CustomStore, DataSource, EdmLiteral, EndpointSelector, LocalStore, ODataContext, ODataStore, applyChanges, base64_encode, compileGetter, compileSetter, errorHandler, isGroupItemsArray, isItemsArray, isLoadResultObject, keyConverters, query, setErrorHandler } from 'devextreme/common/data';
import { exportDataGrid, exportPivotGrid } from 'devextreme/common/export/excel';
import { exportDataGrid as exportDataGrid$1, exportGantt } from 'devextreme/common/export/pdf';

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index$b = /*#__PURE__*/Object.freeze({
	__proto__: null,
	AIIntegration: AIIntegration
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index$a = /*#__PURE__*/Object.freeze({
	__proto__: null,
	registerGradient: registerGradient,
	registerPattern: registerPattern
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index$9 = /*#__PURE__*/Object.freeze({
	__proto__: null,
	TransitionExecutor: TransitionExecutor,
	animationPresets: animationPresets,
	cancelAnimationFrame: cancelAnimationFrame,
	fx: fx,
	requestAnimationFrame: requestAnimationFrame
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index$8 = /*#__PURE__*/Object.freeze({
	__proto__: null,
	getTimeZones: getTimeZones,
	hideTopOverlay: hideTopOverlay,
	initMobileViewport: initMobileViewport
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index$7 = /*#__PURE__*/Object.freeze({
	__proto__: null,
	off: off,
	on: on,
	one: one,
	trigger: trigger
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index$6 = /*#__PURE__*/Object.freeze({
	__proto__: null,
	formatDate: formatDate,
	formatMessage: formatMessage,
	formatNumber: formatNumber,
	loadMessages: loadMessages,
	locale: locale,
	parseDate: parseDate,
	parseNumber: parseNumber
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index$5 = /*#__PURE__*/Object.freeze({
	__proto__: null,
	Animation: index$9,
	Environment: index$8,
	Events: index$7,
	Localization: index$6
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index$4 = /*#__PURE__*/Object.freeze({
	__proto__: null,
	ArrayStore: ArrayStore,
	CustomStore: CustomStore,
	DataSource: DataSource,
	EdmLiteral: EdmLiteral,
	EndpointSelector: EndpointSelector,
	LocalStore: LocalStore,
	ODataContext: ODataContext,
	ODataStore: ODataStore,
	applyChanges: applyChanges,
	base64_encode: base64_encode,
	compileGetter: compileGetter,
	compileSetter: compileSetter,
	errorHandler: errorHandler,
	isGroupItemsArray: isGroupItemsArray,
	isItemsArray: isItemsArray,
	isLoadResultObject: isLoadResultObject,
	keyConverters: keyConverters,
	query: query,
	setErrorHandler: setErrorHandler
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index$3 = /*#__PURE__*/Object.freeze({
	__proto__: null,
	exportDataGrid: exportDataGrid,
	exportPivotGrid: exportPivotGrid
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index$2 = /*#__PURE__*/Object.freeze({
	__proto__: null,
	exportDataGrid: exportDataGrid$1,
	exportGantt: exportGantt
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index$1 = /*#__PURE__*/Object.freeze({
	__proto__: null,
	Excel: index$3,
	Pdf: index$2
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

var index = /*#__PURE__*/Object.freeze({
	__proto__: null
});

/*!
 * devextreme-angular
 * Version: 25.1.6
 * Build date: Mon Oct 13 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */

/**
 * Generated bundle index. Do not edit.
 */

export { index$b as AiIntegration, index$a as Charts, index$5 as Core, index$4 as Data, index$1 as Export, index as Grids };
