// devextreme-schematics/src/migrate-config-components/index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.migrateConfigComponents = void 0;
const template_migrator_1 = require("./template-migrator");
const deprecated_config_map_json_1 = __importDefault(require("./mappings/deprecated-config-map.json"));
function migrateConfigComponents(options = {}) {
    const include = normalizeGlobOption(options.include, ['**/*.html']);
    const dryFlag = normalizeBoolean(options.dry);
    const scriptGlobs = normalizeGlobOption(options.scriptInclude, ['**/*.ts', '**/*.js']);
    return (tree, ctx) => __awaiter(this, void 0, void 0, function* () {
        ctx.logger.info(`[config-migrator] Starting…`);
        const hostMap = deprecated_config_map_json_1.default;
        const processedMapping = Object.entries(hostMap).map(([hostComponentName, map]) => {
            var _a;
            const hostSelector = (_a = map._hostSelector) !== null && _a !== void 0 ? _a : componentToSelectorGuess(hostComponentName);
            const configMap = Object.fromEntries(Object.entries(map).filter(([k]) => k !== '_hostSelector'));
            return { hostSelector, configMap };
        });
        // External HTML templates
        yield (0, template_migrator_1.applyHostAwareTemplateMigrations)(tree, {
            includeGlobs: include,
            rules: processedMapping
        }, { dryRun: dryFlag, logger: ctx.logger });
        // Inline templates inside component decorators (ts/js)
        yield (0, template_migrator_1.applyInlineComponentTemplateMigrations)(tree, {
            includeGlobs: [], rules: processedMapping
        }, { dryRun: dryFlag, logger: ctx.logger }, scriptGlobs);
        ctx.logger.info(`[config-migrator] Done.`);
    });
}
exports.migrateConfigComponents = migrateConfigComponents;
// Fallback if _hostSelector is missing: "DxFooBarComponent" -> "dx-foo-bar"
function componentToSelectorGuess(componentName) {
    const core = componentName.replace(/Component$/, '').replace(/^Dx/, 'dx-');
    return core.replace(/([a-z0-9])([A-Z])/g, '$1-$2').toLowerCase();
}
function normalizeGlobOption(value, fallback) {
    if (Array.isArray(value)) {
        return value.filter(Boolean);
    }
    if (typeof value === 'string' && value.trim()) {
        const trimmed = value.trim();
        const inner = trimmed.startsWith('[') && trimmed.endsWith(']')
            ? trimmed.slice(1, -1)
            : trimmed;
        return inner
            .split(',')
            .map(segment => segment.trim())
            .filter(Boolean);
    }
    return fallback.slice();
}
function normalizeBoolean(value) {
    if (typeof value === 'string') {
        return value.toLowerCase() === 'true';
    }
    return !!value;
}
